lib/actions/appleMusic.js

```diff
@@ -38,8 +52,127 @@ function appleMusic(player, values) {
   
   var uri = uriTemplates[type].format({id: encodeURIComponent(trackID)});
 
+  // Parse the iTunes search results and load queue with tracks.  The search name can be an artist name OR a song title  
+  var handleiTunesResponse = function (error, tracksJson) {
+    if (!error) {
+      try { 
+        tracksJson = JSON.parse(tracksJson);
+        console.log("Found " + tracksJson.resultCount + " tracks for " + trackID);
+        if (tracksJson.resultCount > 0) {
+        
+          // Find the first streamable track
+          var trkNum; 
+          for (trkNum=0;(trkNum < tracksJson.resultCount) && !tracksJson.results[trkNum].isStreamable ;trkNum++) {
+          }
+
+          // There is at least one streamable track  
+          if (trkNum < tracksJson.resultCount) {
+            var prevArtist = tracksJson.results[trkNum].artistName;
+            var prevTrack = tracksJson.results[trkNum].trackName;
+            var artistCount = 1;
+            var trackCount = 1;
+            
+            // Determine if the search results are many tracks for an artist OR a specific song title.
+            for (var i=trkNum; i < tracksJson.resultCount;i++) {
+              var track = tracksJson.results[i];
+              
+              if (track.artistName != prevArtist) {
+                artistCount++;
+                prevArtist = track.artistName;
+              }
+              if (track.trackName != prevTrack) {
+                trackCount++;
+                prevTrack = track.trackName;
+              }  
+            }
+          
+            // If there are more track differences that artist differences, then load all of the artist's tracks into the queue, ELSE play just the first track now (iTunes returns the most popular track first) 
+            if (trackCount > artistCount) {
+              player.coordinator.removeAllTracksFromQueue();
+              setTimeout(function() {
+                // Add the first track and start it playing
```

Comment:
Run this in the callback of removeAllTracksFromQueue instead, so you don't have to estimate the wait. The callback should be invoked when the queue has been cleared.